packages/swayze/test/core/controller/table/swayze_header_drag_state_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:swayze/src/core/controller/table/header_state.dart';
+import 'package:swayze_math/swayze_math.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Swayze header drag state', () {
+    test('is allowed to drop', () {
+      expect(
+        const SwayzeHeaderDragState(
+          headers: Range(0, 5),
+          position: Offset(10, 0),
+          dropAtIndex: 10,
+        ).isDropAllowed,
+        isTrue,
+      );
+      expect(
+        const SwayzeHeaderDragState(
+          headers: Range(0, 5),
+          position: Offset(10, 0),
+          dropAtIndex: 6,
+        ).isDropAllowed,
+        isTrue,
+      );
+      expect(
+        const SwayzeHeaderDragState(
+          headers: Range(10, 11),
+          position: Offset(10, 0),
+          dropAtIndex: 9,
+        ).isDropAllowed,
+        isTrue,
+      );
+    });
+
+    test('is not allowed to drop', () {
+      expect(
+        const SwayzeHeaderDragState(
+          headers: Range(0, 5),
+          position: Offset(10, 0),
+          dropAtIndex: 2,
+        ).isDropAllowed,
+        isFalse,
+      );
+      expect(
+        const SwayzeHeaderDragState(
+          headers: Range(10, 11),
+          position: Offset(10, 0),
+          dropAtIndex: 10,
+        ).isDropAllowed,
+        isFalse,
```

Comment:
We have a test for the left boundary, should we add another one for the right boundary?

Reply:
not sure if I follow

Reply:
yes, but that would be `end - 1`, because our range is non inclusive.